app/code/Magento/Inventory/Model/Stock/SearchCriteria/CollectionProcessor/StockFilterProcessor/StockNameFilter.php

```diff
@@ -0,0 +1,39 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\Inventory\Model\Stock\SearchCriteria\CollectionProcessor\StockFilterProcessor;
+
+use Magento\Framework\Api\Filter;
+use Magento\Framework\Api\SearchCriteria\CollectionProcessor\FilterProcessor\CustomFilterInterface;
+use Magento\Framework\Data\Collection\AbstractDb;
+
+class StockNameFilter implements CustomFilterInterface
```

Comment:
I'm not sure that we need to bind this filter only for name...
You can reuse this filter for another fields... or another entities, for exampe we have the same problem in source grid

My propouse is using something like `FieldSearchFilter` for class name

Reply:
This was meant to be used only for this specific grid. For a more generic solution won't be better something like #1285 that operates at framework level?